spec/services/batch_runner_spec.rb

```diff
@@ -40,9 +77,147 @@
     end
   end
 
-  # describe 'read' do
-  # end
-  #
-  # describe 'enqueue' do
-  # end
+  describe 'read' do
+    let(:batch_runner) { described_class.new(batch: batch) }
+    let(:batch) { FactoryBot.build(:initialized_batch) }
+    let(:submitter_email) { batch.submitter_email }
+
+    it 'raises ArguementError if batch not persisted yet' do
+      expect { batch_runner.read }.to raise_error(ArgumentError)
+    end
+
+    context 'with batch initialized' do
+      let(:batch) { FactoryBot.create(:initialized_batch) }
+      let(:config) { instance_double(Hyrax::BatchIngest::IngestTypeConfig, reader: reader_class) }
+      let(:reader_class) { double("ReaderClass") }
+      let(:reader) { double("BatchReader") }
```

Comment:
@cjcolvar what does calling `double` on a string do here? Why not just use a string?

Reply:
I started with `instance_double` but switched to `double`.  Now that the other PRs were merged I'll switch it back.